scripts/travis-pr-title-lint

```diff
@@ -0,0 +1,181 @@
+
+    var req = https.get({
+      host: 'github.com',
+      path: pullRequestURLPath,
+      headers: {
+        'User-Agent': 'GitHub... your API can be very annoying ;-)',
+      },
+    }, function(response) {
+      if (response.statusCode < 200 || response.statusCode > 299) {
+        reject(new Error(`getPullRequestTitle got an unexpected statusCode: ${response.statusCode}`));
+        return;
+      }
+
+      var body = '';
+      response.on('data', function(data) {
+        try {
+          body += data;
+
+          // Once we get the closing title tag, we can read
+          // the pull request title and
+          if (body.includes('</title>')) {
+            response.removeAllListeners('data');
+            response.emit('end');
+
+            var titleStart = body.indexOf('<title>');
+            var titleEnd = body.indexOf('</title>');
+
+            // NOTE: page slice is going to be something like:
+            // "<title> PR title by author · Pull Request #NUM · mozilla/web-ext · GitHub"
+            var pageTitleParts = body.slice(titleStart, titleEnd)
+              .replace('<title>', '')
+              .split(' · ');
+
+            // Check that we have really got the title of a real pull request.
+            var expectedPart1 = `Pull Request #${TRAVIS_PULL_REQUEST}`;
+
+            if (pageTitleParts[1] === expectedPart1) {
+              // Remove the "by author" part.
+              var prTitleEnd = pageTitleParts[0].lastIndexOf(' by ');
+              resolve(pageTitleParts[0].slice(0, prTitleEnd));
+            } else {
+              console.log('DEBUG getPullRequestTitle response:', body);
+
+              reject(new Error('Unable to retrieve the pull request title'));
+            }
+
+            req.abort();
+          }
+        } catch (err) {
+          reject(err);
+          req.abort();
+        }
+      });
+      response.on('error', function(err) {
+        console.error('Failed during pull request title download: ', err);
+        reject(err);
+      });
+    });
+  });
+}
+
+function lintMessage(message) {
+  if (!message) {
+    throw new Error('Unable to lint an empty message.');
+  }
+
+  const res = shelljs.exec(`echo ${message} | commitlint`, {silent: true});
+
+  if (res.code !== 0) {
+    throw res.stdout;
+  }
+
+  return res.stdout.trim();
+}
+
+async function runChangelogLinting() {
+  try {
+    const commits = getGitBranchCommits();
+    let message;
+
+    if (commits.length === 1) {
+      console.log('There is only one commit in this pull request,',
+                  'we are going to check the single commit message...');
+      message = getGitCommitMessage(commits[0]);
+    } else {
+      console.log('There is more than one commit in this pull request,',
+                  'we are going to check the pull request title...');
+
+      message = await getPullRequestTitle();
+    }
+
+    lintMessage(message);
+  } catch (err) {
+    var errMessage = err.stack ? err.stack : err;
```

Comment:
If there is no stack on an error object, then `err` may be an (error) object, and at the next line `errorMessage.trim()` would throw. Consider using the following:
```
var errMessage = `${err.stack || err}`.trim();
```